dozer/cogs/tba.py

```diff
@@ -37,100 +41,270 @@ def __init__(self, bot):
+                           if year is not None else "This team hasn't won any awards...yet.")
+
+    awards.example_usage = """
+    `{prefix}tba awards 1114` - list all the awards team 1114 Simbotics has ever gotten.
+    """
+
     @tba.command()
     async def raw(self, ctx, team_num: int):
         """
         Get raw TBA API output for a team.
         This command is really only useful for development.
         """
-        team_data = self.parser.get_team(team_num)
         try:
-            getattr(team_data, "Errors")
-        except tbapi.InvalidKeyError:
-            e = discord.Embed(color=blurple)
+            team_data = await self.session.team(team_num)
+            e = discord.Embed(color=discord.Color.blurple())
             e.set_author(name='FIRST® Robotics Competition Team {}'.format(team_num),
                          url='https://www.thebluealliance.com/team/{}'.format(team_num),
                          icon_url='https://frcavatars.herokuapp.com/get_image?team={}'.format(team_num))
-            e.add_field(name='Raw Data', value=team_data.flatten())
+            e.add_field(name='Raw Data', value="``` \n {}```".format(pformat(team_data.__dict__)))
             e.set_footer(text='Triggered by ' + ctx.author.display_name)
             await ctx.send(embed=e)
-        else:
+        except aiotba.http.AioTBAError:
             raise BadArgument('Team {} does not exist.'.format(team_num))
 
     raw.example_usage = """
     `{prefix}tba raw 4150` - show raw information on team 4150, FRobotics
     """
 
+    class TeamData:
+        """polyfill data class used to abstract team location data from frc/ftc"""
+        country: str
+        state_prov: str
+        city: str
+
+    @command()
+    @bot_has_permissions(embed_links=True)
+    async def weather(self, ctx, team_num: int, team_program="frc"):
+        """Finds the current weather for a given team."""
+
+        if team_program.lower() == "frc":
+            try:
+                td = await self.session.team(team_num)
+            except aiotba.http.AioTBAError:
+                raise BadArgument('Team {} does not exist.'.format(team_num))
+        elif team_program.lower() == "ftc":
+            res = json.loads(await self.bot.cogs['TOA'].parser.req("team/{}".format(str(team_num))))
+            if not res:
+                raise BadArgument('Team {} does not exist.'.format(team_num))
+            td_dict = res[0]
+            td = self.TeamData()
+            td.__dict__.update(td_dict)
+        else:
+            raise BadArgument('`team_program` should be one of [`frc`, `ftc`]')
+
+        units = 'm'
+        if td.country == "USA":
+            td.country = "United States of America"
+            units = 'u'
+        e = discord.Embed(title=f"Current weather for {team_program.upper()} Team {team_num}:")
+        e.set_image(url="https://wttr.in/" + urlquote(f"{td.city}+{td.state_prov}+{td.country}_0_{units}.png"))
```

Comment:
This should be a `str.format` call.

Reply:
any reason over an f-string?

Reply:
Curly brackets aren't allowed inside expressions in f-strings, so f-strings that contain f-strings like this would be raise syntax errors.